### ISSUE TITLE
Endpoint to populate airtable with data from the FE 'Team Treehouse Licenses' form

### ISSUE THREAD

**Issue by hpjaj** [OperationCode/operationcode_backend#288]
# Feature
## Why is this feature being added?
The FE will be creating a form around Team Treehouse Licenses.  This form will only be accessible to ID.me verified users.  This data needs to ultimately be populated in Airtable in a new airtable.

## Associated Issue

https://github.com/OperationCode/operationcode_frontend/issues/890

## What should your feature do?

- [ ] create a new airtable in Airtable for the data (@hpjaj will take care of this)
- [ ] create a `POST` endpoint that hits Airtables API and populates the new airtable with the passed data
- [ ] validation that the passed `user` is verified
- [ ] test coverage
- [ ] updates our API docs
- [ ] collaboration with the FE team on the endpoint's schema (i.e. what they are sending to the BE)

**Comment by hpjaj** [comment]
@kylemh ! 

Can you please describe:

1.  Precisely what data this FE form will be passing to the BE
2.  If that data will/will not be a 1:1 mapping to columns in an Airtable airtable?
3.  Anything else that I am missing in the description re: acceptance criteria for the FE

**Comment by ag-kylemh** [comment]
Instead of making this just about Tree House licenses, perhaps it could be our scholarship application page? I envision a drop-down representing all the available scholarships/resources one can sign up with.

Regardless, our prior discussion was relating to administrative tools, so this would be slightly different. Luckily, whatever we do here can be used there.

I presume the following data will get sent to the back-end from such a form:

**Name** (String)
**Branch** (String or Number - your choice)

Anything else would be up to @AshTemp

**Comment by kylemh** [comment]
Instead of making this just about Tree House licenses, perhaps it could be our scholarship application page? I envision a drop-down representing all the available scholarships/resources one can sign up with.

Regardless, our prior discussion was relating to administrative tools, so this would be slightly different. Luckily, whatever we do here can be used there.

I presume the following data will get sent to the back-end from such a form:

**Name** (String)
**Branch** (String or Number - your choice)

Anything else would be up to @AshTemp

**Comment by AshTemp** [comment]
Treehouse is the only scholarship we have right now....The idea is good tho.

**Comment by hpjaj** [comment]
@kylemh - You're right, I did use the admin dashboard as the example, but actually I was more broadly thinking exactly this.  Forms & tables, with varied access.  Be that access "admin" access, or "verified user" access, or "board member" access...whatever we choose.  We can create any roles we choose.

But for this example let's say it is for ID.me verified users.  

- These verified users can access this form on the frontend
- When they submit the form, the changes are saved to our db (not Airtable)
- The FE has a table that displays the data from the db, that only user's with xyz permissions can see.  And we can set that level of access to whatever we choose.

I think we should go this route for all the benefits I mentioned in [my thoughts around forms & tables for software devs](https://gist.github.com/hpjaj/0477fabd6287917e62551d05098dcda9).

I also think that since we have this as an option, I would prefer spending our time and resources on investing in and building out our application, versus deepening our dependency on a third party service such as Airtable.

You down?

**Comment by AshTemp** [comment]
Whoa, wait a minute. We are using airtable for a multitude of reasons. The big ones being the mentors can publicly see who still needs a mentor. Others include the fact that the bot can not only read the requests but also write to the table to include who is mentoring an individual. So unless you’re going to also include Allen in this discussion to find out what needs to happen so that the bot still works don’t make too many plans to not use airtable. We’ve already automated a huge part of requesting and fulfilling a mentor, let’s not go backwards.

**Comment by hpjaj** [comment]
Right, this is not for mentors or the mentor process.  This is for Team Treehouse (or scholarship applications).

**Comment by AshTemp** [comment]
Oh, would still need a quick searchable way to find that and teamtree house is basically going to be a spreadsheet so airtable is perfect for that.

**Comment by hpjaj** [comment]
Airtable is one option.  What I am describing accomplishes the same end goal, with many additional benefits.

**Comment by AshTemp** [comment]
Before we make big changes you need to talk with Nell.

**Comment by hpjaj** [comment]
Agreed!  @nellshamrell was part of my original discussion on the topic.  

I just implemented the concept of "roles" in PR https://github.com/OperationCode/operationcode_backend/pull/290.  This, in conjunction with Pundit, provides varied access to forms, links, tables, etc.  

But for sure, @nellshamrell can you pls share your thoughts?

**Comment by nellshamrell** [comment]
Reading through the backscroll now :)

**Comment by nellshamrell** [comment]
Here are my thoughts:

I think everyone (including me) is in agreement that the mentorship process should stay in Airtable.

What I'm hearing here, though, is that we'd like to use our own database to track scholarship information (starting with Tree House).  I believe this would not alter any current behavior - this is entirely new behavior.  Is this correct?  Or do we currently track scholarship information in Airtable (I know we did at one point, I don't know if we do now)?

If we do not, then I think we should go forward with the plan to have it in our own database.  The reason is I think this will provide more learning opportunities for new developers to our open source projects - creating forms and views that create, display, and manipulate data is a great beginning coding project.

**Comment by AshTemp** [comment]
I am tracking exactly 1 entry in airtable at the moment, the other
scholarship stuff I'm handling in Google forms so nothing will be affected.

**Comment by dmarchante** [comment]
@AshTemp @nellshamrell can we revisit the direction we want this to go? Are we still looking to move away from AirTable, and track this ourselves?

**Comment by dmarchante** [comment]
@AshTemp @nellshamrell @hollomancer @JennWeideman 
We need to have a discussion around this as I think we need a db for a variety of reason not just scholarships, and we will have to update the original issue as well.
